src/context.js

```diff
@@ -50,13 +56,48 @@ function EvolvContext(store) {
     }
   }
 
+  function mergePersistedContext(contextKey, context, storage) {
+    const persistedContext = storage.getItem(contextKey);
+    if (persistedContext) {
+      const parsedContext = JSON.parse(persistedContext);
+      if (context) {
+        return objects.deepMerge(context, parsedContext);
+      }
+
+      return parsedContext;
+    }
+
+    return context || {};
+  }
+
+  function mergeAllPersistedStorage(contextKey, context) {
+    if (window === undefined || !window.localStorage) {
+      return context || {};
+    }
+
+    const localStorageContext = mergePersistedContext(contextKey, context, window.localStorage);
+    return mergePersistedContext(contextKey, localStorageContext, window.sessionStorage);
+  }
+
+  function persistValue(key, value, local, sessionOnly) {
+    const storage = sessionOnly ? window.sessionStorage : window.localStorage;
+    const persistedContext = storage.getItem(local ? localStorageKey : remoteStorageKey);
+    const context = persistedContext ? JSON.parse(persistedContext) : {};
+    objects.setKeyToValue(key, value, context);
+    storage.setItem(local ? localStorageKey : remoteStorageKey, JSON.stringify(context));
+  }
+
   this.initialize = function(_uid, _remoteContext, _localContext) {
     if (initialized) {
       throw new Error('Evolv: The context is already initialized');
     }
     uid = _uid;
-    remoteContext = _remoteContext ? objects.deepClone(_remoteContext) : {};
-    localContext = _localContext ? objects.deepClone(_localContext) : {};
+
+    localStorageKey = 'evolv_' + this.uid + '_local_context';
+    remoteStorageKey = 'evolv_' + this.uid + '_remote_context';
+
+    localContext = mergeAllPersistedStorage(localStorageKey, _localContext);
+    remoteContext = mergeAllPersistedStorage(remoteStorageKey, _remoteContext);
```

Comment:
The behavior, I believe, that `initialize()` would have here is counterintuitive. It looks like the values that a user passed as `localContext` or `remoteContext` to `initialize()` could be overwritten by the values from storage (the second argument in `deepMerge()` wins). Is that desirable?

Maybe I don't understand the use case for persisting context values?

Reply:
This would be an edge case, and preference could be argued either way. Treating the parameters as defaults and merging with the stored value taking precedence avoided reconciliation logic, as this bypasses the functionality that would update the storage with new values.